compiler/rustc_ast/src/expand/autodiff_attrs.rs

```diff
@@ -150,7 +160,7 @@ pub fn valid_ty_for_activity(ty: &P<Ty>, activity: DiffActivity) -> bool {
     if matches!(activity, Const) {
         return true;
     }
-    if matches!(activity, Dual | DualOnly) {
+    if matches!(activity, Dual | DualOnly | Dualv | DualvOnly) {
```

Comment:
Maybe add a `is_dual_or_const` function and use it everywhere where these four variants are handled the same